include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -68,26 +68,26 @@ arena_prof_info_get(tsd_t *tsd, const void *ptr, alloc_ctx_t *alloc_ctx,
 }
 
 JEMALLOC_ALWAYS_INLINE void
-arena_prof_tctx_set(tsd_t *tsd, const void *ptr, alloc_ctx_t *alloc_ctx,
-    prof_tctx_t *tctx) {
+arena_prof_tctx_set_not_sampled(tsd_t *tsd, const void *ptr,
+    alloc_ctx_t *alloc_ctx) {
 	cassert(config_prof);
 	assert(ptr != NULL);
 
 	/* Static check. */
 	if (alloc_ctx == NULL) {
 		extent_t *extent = iealloc(tsd_tsdn(tsd), ptr);
 		if (unlikely(!extent_slab_get(extent))) {
-			large_prof_tctx_set(extent, tctx);
+			large_prof_tctx_reset(extent);
 		}
 	} else {
 		if (unlikely(!alloc_ctx->slab)) {
-			large_prof_tctx_set(iealloc(tsd_tsdn(tsd), ptr), tctx);
+			large_prof_tctx_reset(iealloc(tsd_tsdn(tsd), ptr));
 		}
 	}
 }
 
 static inline void
-arena_prof_tctx_reset(tsd_t *tsd, const void *ptr, prof_tctx_t *tctx) {
+arena_prof_tctx_reset(tsd_t *tsd, const void *ptr) {
```

Comment:
Just realized, the above two functions (reset and set not sampled) are doing the same thing, just one is more optimized with alloc_ctx. I’d say keep the reset name, with the alloc_ctx implementation.

Reply:
Right. Changed `prof_tctx_reset()` to `prof_tctx_reset_sampled()` and changed `prof_tctx_set_not_sampled()` to `prof_tctx_reset()`.

Reply:
If they are doing the same thing, maybe merge the two? Or merge the underlying impl.

Reply:
The optimized version just avoids some checking. I think merging the two would essentially end up reducing to the full version and getting rid of the optimization. Is this the intention?

Reply:
We can keep the optimized one, i.e. `arena_prof_tctx_reset` still takes an additional alloc_ctx. `arena_prof_tctx_reset_sampled` can simply call the former and pass NULL alloc_ctx.

Reply:
Right, that's one of the checks. The other is the slab check, which the optimized version doesn't do. If we "squash" the layers, the optimized one basically just contain a single set operation, and it's used whenever the slab check can already been inferred.

Reply:
Ah I see. Missed that part. Never mind.